examples/fp_delegation.cpp

```diff
@@ -123,22 +123,27 @@ class delegating_sender {
     }
   };
 
+  using SchedSender =
+      schedule_result_t<get_scheduler_result_t<timed_single_thread_context&>>;
+
+  template <typename Receiver>
+  using operation =
+      delegating_operation<
+          connect_result_t<
+              schedule_result_t<get_scheduler_result_t<Receiver>>,
+              Receiver>,
+          LocalContextType<connect_result_t<SchedSender, Receiver>>>;
+
   template <typename Receiver>
-  auto connect(Receiver&& receiver) {
+  operation<Receiver> connect(Receiver&& receiver) {
     // Attempt to reserve a slot otherwise delegate to the downstream scheduler
+    using LC = LocalContextType<connect_result_t<SchedSender, Receiver>>;
+    using op = operation<Receiver>;
 
-    using LC = LocalContextType<remove_cvref_t<decltype(unifex::connect(
-      unifex::schedule(context_->single_thread_context_.get_scheduler()),
-      (Receiver&&)receiver))>>;
-    using op = delegating_operation<
-        remove_cvref_t<decltype(unifex::connect(
-          unifex::schedule(unifex::get_scheduler(std::as_const(receiver))),
-          (Receiver&&)receiver))>,
-        LC>;
     if(context_->reserve()) {
       auto local_op = [&receiver, context = context_]() mutable {
         return LC{unifex::connect(
-          unifex::schedule(context->single_thread_context_.get_scheduler()),
+          unifex::schedule(unifex::get_scheduler(context->single_thread_context_)),
```

Comment:
This seems to be the only use of `get_scheduler(context)` which relies on this dispatching to the member `.get_scheduler()` method.

This change should probably be moved out to a separate PR so it can be considered on its own. It doesn't seem to be integral to the main thrust of this PR, which is to fix reduce_stream constraints.

If we do decide that this is the direction we want to go (ie. to allow using `unifex::get_scheduler()` to obtain the scheduler from an execution context) then the alternative to dispatching to member function would be to replace the `.get_scheduler()` methods on the execution context objects with a tag_invoke-based customisation of the existing `unifex::get_scheduler()` CPO and update all calls to use the CPO instead.

Reply:
I can factor it out.